tests/Common/TriggersTest.php

```diff
@@ -339,4 +339,60 @@ public function testTokenWithExpiration()
             'tableIds' => [''],
         ]);
     }
+
+    public function testTriggersRestrictionsForReadOnlyUser()
+    {
+        $expectedError = 'Trigger manipulation is restricted for your user role "readOnly".';
+        $readOnlyClient = $this->getClientForToken(STORAGE_API_READ_ONLY_TOKEN);
+
+        $table1 = $this->createTableWithRandomData("watched-1");
+        $newTokenId = $this->_client->createToken(
+            (new TokenCreateOptions())
+            ->addBucketPermission($this->getTestBucketId(), TokenAbstractOptions::BUCKET_PERMISSION_READ)
+        );
+
+        $options = [
+            'component' => 'orchestrator',
+            'configurationId' => 123,
+            'coolDownPeriodMinutes' => 10,
+            'runWithTokenId' => $newTokenId,
+            'tableIds' => [
+                $table1,
+            ],
+        ];
+
+        $trigger = $this->_client->createTrigger($options);
+        $triggers = $this->_client->listTriggers();
+
+        $this->assertSame($triggers, $readOnlyClient->listTriggers());
+
+        try {
+            $readOnlyClient->createTrigger($options);
+            $this->fail('Triggers API POST request should be restricted for readOnly user');
+        } catch (ClientException $e) {
+            $this->assertSame(403, $e->getCode());
+            $this->assertSame('accessDenied', $e->getStringCode());
+            $this->assertSame($expectedError, $e->getMessage());
+        }
+
+        try {
+            $readOnlyClient->updateTrigger($trigger['id'], ['configurationId' => 987]);
+            $this->fail('Files API PUT request should be restricted for readOnly user');
```

Comment:
Nema byt tu trochu ina hlaska ?

Reply:
opravil jsem ty hlasky, to nevim jak se mi povedlo udelat takovyhle dvojkombo v jednom testu :D